### DISCUSSION TITLE
[2023-02-13] Library & Engine Team Update.

### DISCUSSION THREAD

**Discussion by jdunkerley** [enso-org/enso#5647] [Weekly Demo Day]
- James: Fix for serializing large integers and display in IDE. #5592 
- James: Fix for `Date`, `Time_Of_Day` and `Date_Time` in table viz. #5592 
- James: Removed warning on empty separator for concatenation. #5592
- James: Fix for issue with `List<Value>` in the Table. #5592 
- James: Removing Column and simpler selector API. #5646
- James: Excel Workbook and new File_Format for images and SQLite. #5646
- James: Updated Vector_Editor Widget. #5646
- James: Prototype work for connecting to a directory.
- Radek: `Table.aggregate` returning a hard error if an invalid aggregation. #5579
- Radek: Review and improving warnings on `Table.filter`. #5605 
- Radek: Add problem handling (like Arithmetic Error) to derived columns / expressions. #5605 
- Radek: Investigating issues from personal Enso project with delimited reading and writing.
- Greg: New simpler `Text.toDisplayString` implementation. #5045 
- Hubert: Fix so that Polyglot values are recognized as Enso types in the IDE. #4111
- Hubert: Fix for fully qualified names causing the engine to crash. #5600
- Hubert: Working to improve import so only requested modules are processed. 
- Hubert: Fix infinite loop when any unexpected crash happens during serialization https://github.com/enso-org/enso/pull/5591
- Jaroslav: Investigated start up performance issues of Language Server and Engine and agreed plan forward.
- Jaroslav: Fixed issue with type pattern matching matching the type itself. 
- Jaroslav: Proof of concept for `Regex.compile` ensuring only single compile.
- Jaroslav: Proof of concept for `Text.replace` with special replacement strings. https://github.com/enso-org/enso/compare/develop...JaroslavTulach:wip/jtulach/TRegex_181348590
- Jaroslav: New sbt shorthand `runEngineDistribution` for running the CLI within sbt with arguments. #5609
- Pavel M: Completed and merged custom equality and hash code. #4067
- Pavel M: Consistent ordering for `<`, `>`, `<=` and `>=`. #4067
- Pavel M: Upgrading to GraalVM 22.3.1 - fixes chrome debugger issue. #5602
- Dmitry: Fix for expressions not having a method pointer if updated before suggestion database. #5578 
- Dmitry: Added method pointer for constructor expressions. #5593 
- Dmitry: Working to pre-compute the suggesion DB during build time.
- Pawel G: Working to add what is needed for nested expression.
- Pawel G: Support for drop downs on in-fix functions.
- Pawel G: Design work for the Vector Editor widget.

**Comment by jdunkerley** [comment]
cc: @radeusgd @GregoryTravis @hubertp @JaroslavTulach @Akirathan @4e6 @Frizi @kustosz

**Comment by radeusgd** [comment]
Some late screenshots showing the new error handling in expressions
<img width="588" alt="image" src="https://user-images.githubusercontent.com/1436948/218531923-c7febcf2-4ba1-46c9-9f08-d21e07081ca7.png">
<img width="700" alt="image" src="https://user-images.githubusercontent.com/1436948/218532631-b53aa58a-8c72-4999-b317-25e8ecc12666.png">
<img width="746" alt="image" src="https://user-images.githubusercontent.com/1436948/218533853-1e0a82d8-7440-4f0a-a9ce-c7b097f38632.png">
